### PR TITLE
Adjust the subtext of the replica set header based on annotations

### PR DESCRIPTION
**Pull request by sorosoro2**
Waiting for the job name variable to be finalised. I'll replace the place holder with that

image

### CODE REVIEW

File: src/WebUI/Utils.ts [hunk 1]
```diff
@@ -13,10 +13,19 @@ import { ILabelModel } from "azure-devops-ui/Label";
 
 const pipelineNameAnnotationKey: string = "azure-pipelines/pipeline";
 const pipelineExecutionIdAnnotationKey: string = "azure-pipelines/execution";
```
Comment (madhuv-msft) [inline src/WebUI/Utils.ts]: for consistency should we rename this to pipelineRunNameAnnotationKey

File: src/WebUI/WorkLoads/WorkloadDetails.scss [hunk 2]
```diff
@@ -1,3 +1,7 @@
 .workload-details-card .workload-full-details-table {
     padding-top: 10px;
+}
+
+.runs-bullet {
+    margin: 0 6px;
 }
```
Comment (madhuv-msft) [inline src/WebUI/WorkLoads/WorkloadDetails.scss]: is this too generic, or are we okay?

File: src/WebUI/WorkLoads/WorkloadDetails.tsx [hunk 3]
```diff
@@ -170,10 +170,53 @@ export class WorkloadDetails extends BaseComponent<IWorkloadDetailsProperties, I
...
+            const pipelineDetails = Utils.getPipelineDetails(metadata.annotations);
             const tableItems: IWorkLoadDetailsItem[] = [{ podTemplate: this.props.podTemplate, parentMetaData: metadata, selector: this.props.selector }];
             const agoTime = Date_Utils.ago(new Date(metadata.creationTimestamp), Date_Utils.AgoFormat.Compact);
 
+            const getSubtext = (pipelineDetails: IMetadataAnnotationPipeline): React.ReactNode => {
+                const first = pipelineDetails.jobName
+                    ? localeFormat(Resources.ServiceCreatedWithPipelineText, agoTime, pipelineDetails.jobName)
+                    : localeFormat(Resources.CreatedAgo, agoTime);
+
+                const runElement = pipelineDetails.runName
```
Comment (madhuv-msft) [inline src/WebUI/WorkLoads/WorkloadDetails.tsx]: should we open in a new tab or in the same window? also we have to make sure the text is blank and on hover or focus only it turn to blue

File: src/WebUI/WorkLoads/WorkloadDetails.tsx [hunk 4]
```diff
@@ -170,10 +170,53 @@ export class WorkloadDetails extends BaseComponent<IWorkloadDetailsProperties, I
...
+                    // Trying to format the resource string with {0} possibly being a link.
+                    // First put the pipeline name, and in the place on run, put {0}
+                    // Pipeline name might have surrounding quotes
+                    const runText = localeFormat(Resources.RunInformationForWorkload, "{0}", pipelineDetails.pipelineName.replace(/^"|"$/g, ""));
+
+                    // Find index of the run placeholder
+                    const indexOfRun = runText.indexOf("{0}");
+
+                    second = (<span>
+                        {runText.substring(0, indexOfRun - 1)}
```
Comment (madhuv-msft) [inline src/WebUI/WorkLoads/WorkloadDetails.tsx]: <span> may not be needed, if you do not need to control other items, so use <> </>

File: src/WebUI/WorkLoads/WorkloadDetails.tsx [hunk 5]
```diff
@@ -170,10 +170,53 @@ export class WorkloadDetails extends BaseComponent<IWorkloadDetailsProperties, I
...
+                        {runText.substring(indexOfRun + 3)}
+                    </span>);
+                }
+
+                return (
+                    <span>
+                        {first}
+                        {
+                            second
+                                ? <span>
```
Comment (madhuv-msft) [inline src/WebUI/WorkLoads/WorkloadDetails.tsx]: <span> is not needed, we can use <> </>

**Comment by madhuv-msft** [comment]
please check if any of the annotations are not available what happens?

**Comment by sorosoro2** [comment]
> please check if any of the annotations are not available what happens?

@madhuv-msft , It will show only "Created <ago_time>"

**Comment by madhuv-msft** [comment]
in podoverview we want to show Job details also, would you add that row also please if this data is available for a pod?

File: src/WebUI/WorkLoads/WorkloadDetails.scss (hunk 2 above)
Comment (sorosoro2) [reply src/WebUI/WorkLoads/WorkloadDetails.scss]: should be fine. I don't think anyone will override this accidentally, as it is a span

File: src/WebUI/WorkLoads/WorkloadDetails.tsx (hunk 3 above)
Comment (sorosoro2) [reply src/WebUI/WorkLoads/WorkloadDetails.tsx]: I'll take this in the next PR

**Pull request merged**